File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        mul=1
        for i in nums:
            mul*=i
        ans=[]
        for i in range(len(nums)):
            if nums[i]==0:
                m=1
                for j in range(i):
                    m*=nums[j]
                for k in range(i+1,len(nums)):
                    m*=nums[k]
                ans.append(m)
            else:
                ans.append(mul//nums[i])
        return ans

        
        

File: Data Structures & Algorithms/split-array-largest-sum/submission-0.java
class Solution {
    public int splitArray(int[] nums, int k) {
        int l=nums[0];
        int r=0;
        for(int i:nums){
            l=Math.max(i,l);
            r=r+i;
        }
        int ans=Integer.MAX_VALUE;
        while(l<=r){
            int m=l+(r-l+1)/2;
            if(helper(nums,k,m)){
                r=m-1;
                ans=Math.min(ans,m);
            }else{
                l=m+1;
            }
        }
        return ans;
    }
    public boolean helper(int[] nums , int k , int m){
            int requiredUser=1;
            int sum=0;
            for(int i:nums){
                sum+=i;
                if(sum>m){
                    requiredUser++;
                    sum=i;
                }
            }
            if(requiredUser<=k){
                return true;
            }
            return false;
    }
}